examples/tutorials/cnndm_summarization.py

```diff
@@ -149,21 +149,76 @@ def apply_prefix(task, x):
 #
 # We can define a sequence generator to produce an output sequence based on the input sequence provided. This calls on the
 # model's encoder and decoder, and iteratively expands the decoded sequences until the end-of-sequence token is generated
-# for all sequences in the batch. The `greedy_generator` method shown below uses a greedy search (i.e. expands the sequence
-# based on the most probable next word).
+# for all sequences in the batch. The `generate` method shown below uses a beam search to generate the sequences.
 #
 
 from torch import Tensor
 from torchtext.prototype.models import T5Model
 
 
```

Comment:
One thing to note is that when generating the first tokens of the sequences, `decoder_tokens` has shape `(batch_size, 1)`. Since we are using a beam search, at that first iteration each sequence has `k` many tokens with which to start the sequence. Since the decoder expects `decoder_tokens` to be 2D, with one sequence per row, we treat each beam as its own sequence such that `decoder_tokens` now has shape `(batch_size * beam_size, 2)`, where the first `k` rows are the beams belonging to the original sequence 1, the next `k` rows are the beams belonging to the original sequence 2, etc. Since the decoder requires the encoder outputs, we must also reshape modify the encoder output since it has the output for a `batch_size` number of sequence. Lines `239-244` define a new order where we repeat each encoder output related to each given original sequence `k` times. This is so that that as we pass in each beam as its own sequence, the decoder still has the corresponding encoder output for the sequence.